provisioning/standalone/vscode/README.md

```diff
@@ -0,0 +1,128 @@
+
+    ```Dockerfile
+    INCLUDE+ ./base/Dockerfile
+    ```
+
+3.   **Copy** the starting project files and directories to the root folder of the container
+4.   **Download**, through `apt-get`, all needed dependencies and packages and clean what is not necessary for your use case.
+5.   **Set** the **user** and **group** permissions of the previous files
+6.   *Optionally* **install code-server extensions** through the [**CLI utility**](#install-extensions-from-cli)
+7.   **Switch** to `${USER}`
+8.   **Add** `ENTRYPOINT ./start.sh ` at the end of the file.
+
+### Build Dockerfile
+
+To build an image is necessary to set the **Docker context** in `/provisioning/standalone/vscode` or another parent directory. That's because we are using `INCLUDE+` (see more [**here**](https://github.com/edrevo/dockerfile-plus.git)) and the used context must contain the files used by the **base image** Dockerfile.
+
+This is an example of how to build the **c-cpp vscode** version:
+
+- Go inside the `/provisioning/standalone/vscode` folder
+- Run: `docker build -f ./c-cpp/Dockerfile -t harbor.crownlabs.polito.it/crownlabs-standalone/vscode-c-cpp`
+
+*Optionally*, you can install **codetogether** providing the **build-arg** `CODETOGETHER_ENABLED_ARG` (see more [here](#codetogether))
+
+## Customized images available
+
+This is a list of **ready-to-use images**, built by the **CrownLabs** team :
+```
+harbor.crownlabs.polito.it/crownlabs-standalone/vscode-c-cpp
+harbor.crownlabs.polito.it/crownlabs-standalone/vscode-python
+harbor.crownlabs.polito.it/crownlabs-standalone/vscode-golang
+```
+
+# Advanced Features
+
+## Install extensions from CLI
+
+**Extensions** can be installed from the command line using:
+
+```Dockerfile
+code-server --extensions-dir /config/extensions --install-extension ${EXTENSION_NAME}
+```
+
+The `EXTENSION_NAME` can be either a local `.vsix` file or a reference to an extension on [https://open-vsx.org/](https://open-vsx.org/) marketplace.
+
+## Disable Marketplace
+
+By default **vscode images** allow users to download extensions from the marketplace without limits. To avoid this, you can **disable the marketplace** by adding the `--disable-marketplace` at startup.
+
+For example:
+
+```
+docker run -e CROWNLABS_LISTEN_PORT=8001 -it --rm -p 8001:8001 vscode-c-cpp --disable-marketplace
+```
+
+## Codetogether
+
+**Vscode images** support [**codetogether**](https://www.codetogether.com/) extension and API, which allow **social-coding** between multiple vscode instances (similarly to **[liveshare](https://visualstudio.microsoft.com/it/services/live-share/)** by [**Microsoft®**](https://www.microsoft.com/). This [video](https://youtu.be/l4yTfduxptw) shows what it can do. To enable an image to use **codetogether** is necessary to build it with the `CODETOGETHER_ENABLED_ARG=true` build argument.
```

Comment:
```suggestion
**Vscode images** support [**codetogether**](https://www.codetogether.com/) extension and API, which allow **social-coding** between multiple vscode instances (similarly to **[liveshare](https://visualstudio.microsoft.com/it/services/live-share/)** by [**Microsoft®**](https://www.microsoft.com/). This [video](https://youtu.be/l4yTfduxptw) shows what it can do. To enable an image to use **codetogether** it is necessary to build it with the `CODETOGETHER_ENABLED_ARG=true` build argument.
```